lib/rules/alpha-value-notation/index.js

```diff
@@ -68,21 +79,15 @@ function rule(primary, options, context) {
 
 				if (!isNumber(value) && !isPercentage(value)) return;
 
-				const optionFuncs = {
-					number: {
-						expFunc: isNumber,
-						fixFunc: asNumber,
-					},
-					percentage: {
-						expFunc: isPercentage,
-						fixFunc: asPercentage,
-					},
-				};
-
+				/** @type {'number' | 'percentage'} */
 				let expectation = primary;
 
-				if (optionsMatches(options, 'exceptProperties', decl.prop)) {
-					expectation = Object.keys(optionFuncs).filter((key) => key !== expectation);
+				if (optionsMatches(secondaryOptions, 'exceptProperties', decl.prop)) {
+					if (expectation === 'number') {
+						expectation = 'percentage';
+					} else if (expectation === 'percentage') {
+						expectation = 'number';
+					}
 				}
```

Comment:
[note] `Object.keys()` leads to more inaccurate type (`string[]`), so I've changed the logic.